components/edit-collective/Form.js

```diff
@@ -258,7 +258,7 @@ class EditCollectiveForm extends React.Component {
       // GraphQL schema has address embedded within location
       // mutation expects { location: { address: '' } }
       if (['address', 'country'].includes(fieldname)) {
-        collective.location = collective.location || {};
+        collective.location = { ...collective.location };
```

Comment:
Also fixed this, it's related to the Apollo update (trying to edit an immutable object)